app/code/Magento/Elasticsearch/Model/Indexer/Fulltext/Plugin/Category/Product/Action/Rows.php

```diff
@@ -77,12 +77,15 @@ public function afterExecute(
     ): ActionRows {
         $indexer = $this->indexerRegistry->get(FulltextIndexer::INDEXER_ID);
         if (!empty($entityIds) && $indexer->isScheduled()) {
+            $productIds = [];
+
             foreach ($this->storeManager->getStores() as $store) {
                 $indexTable = $this->getIndexTable((int) $store->getId(), $useTempTable);
-                $productIds = $this->getProductIdsFromIndex($indexTable, $entityIds);
-                if (!empty($productIds)) {
-                    $indexer->reindexList($productIds);
-                }
+                $productIds = array_merge($productIds, $this->getProductIdsFromIndex($indexTable, $entityIds));
+            }
+
+            if (!empty($productIds)) {
+                $indexer->reindexList(array_unique($productIds));
```

Comment:
Let's split all product ids into some small enough pieces (up to 50 products) and reindex this small piece.

Reply:
Out of curiosity: what is the benefit of chunking the amount of productId's up? The original code also didn't do this.

Reply:
We might have a few thousand of products, so it's better to reindex with smaller pieces.